### PR TITLE
[chore] move kafkareceiver to generated lifecycle tests

### PR DESCRIPTION
**Pull request by atoulme**
Related to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27850

### CODE REVIEW

File: receiver/kafkareceiver/factory_test.go [hunk 1]
```diff
@@ -35,9 +35,10 @@ func TestCreateTracesReceiver(t *testing.T) {
 	cfg.ProtocolVersion = "2.0.0"
 	f := kafkaReceiverFactory{tracesUnmarshalers: defaultTracesUnmarshalers()}
 	r, err := f.createTracesReceiver(context.Background(), receivertest.NewNopCreateSettings(), cfg, nil)
+	require.NoError(t, err)
+	err = r.Start(context.Background(), componenttest.NewNopHost())
 	// no available broker
 	require.Error(t, err)
```
Comment (bogdandrutu) [inline receiver/kafkareceiver/factory_test.go:41]: require.Error(t, r.Start(context.Background(), componenttest.NewNopHost()))

File: receiver/kafkareceiver/factory_test.go [hunk 2]
```diff
@@ -123,9 +125,10 @@ func TestCreateLogsReceiver(t *testing.T) {
 	cfg.ProtocolVersion = "2.0.0"
 	f := kafkaReceiverFactory{logsUnmarshalers: defaultLogsUnmarshalers("Test Version", zap.NewNop())}
 	r, err := f.createLogsReceiver(context.Background(), receivertest.NewNopCreateSettings(), cfg, nil)
+	require.NoError(t, err)
+	err = r.Start(context.Background(), componenttest.NewNopHost())
 	// no available broker
 	require.Error(t, err)
```
Comment (bogdandrutu) [inline receiver/kafkareceiver/factory_test.go:131]: same.

File: receiver/kafkareceiver/factory_test.go [hunk 3]
```diff
@@ -79,9 +80,10 @@ func TestCreateMetricsReceiver(t *testing.T) {
 	cfg.ProtocolVersion = "2.0.0"
 	f := kafkaReceiverFactory{metricsUnmarshalers: defaultMetricsUnmarshalers()}
 	r, err := f.createMetricsReceiver(context.Background(), receivertest.NewNopCreateSettings(), cfg, nil)
+	require.NoError(t, err)
+	err = r.Start(context.Background(), componenttest.NewNopHost())
 	// no available broker
 	require.Error(t, err)
```
Comment (bogdandrutu) [inline receiver/kafkareceiver/factory_test.go:86]: same

File: receiver/kafkareceiver/kafka_receiver.go [hunk 4]
```diff
@@ -136,6 +110,38 @@ func (c *kafkaTracesConsumer) Start(_ context.Context, host component.Host) erro
...
+			saramaConfig.Net.ResolveCanonicalBootstrapServers = true
+		}
+		if c.config.ProtocolVersion != "" {
+			version, err := sarama.ParseKafkaVersion(c.config.ProtocolVersion)
+			if err != nil {
+				return err
+			}
+			saramaConfig.Version = version
+		}
+		if err := kafka.ConfigureAuthentication(c.config.Authentication, saramaConfig); err != nil {
```
Comment (bogdandrutu) [inline receiver/kafkareceiver/kafka_receiver.go:136]: Extract all this config conversion in a separate func please.

File: receiver/kafkareceiver/kafka_receiver.go [hunk 5]
```diff
@@ -136,6 +110,38 @@ func (c *kafkaTracesConsumer) Start(_ context.Context, host component.Host) erro
 	if err != nil {
 		return err
 	}
+	if c.consumerGroup == nil {
```
Comment (bogdandrutu) [inline receiver/kafkareceiver/kafka_receiver.go:113]: add a comment about this, sounds kind of random to check for this, but it is for testing.

Comment (atoulme) [reply receiver/kafkareceiver/kafka_receiver.go:113]: It's not just for testing, so I guess we can make this more obvious.
We have had panics on shutdown when the collector doesn't start correctly, where shutdown might get called on receivers that have not started yet. This check ensures that we don't crash if `Shutdown` is called without calling `Start` first.

Comment (bogdandrutu) [reply receiver/kafkareceiver/kafka_receiver.go:113]: Are you sure? Because this is in `Start`

Comment (atoulme) [reply receiver/kafkareceiver/kafka_receiver.go:113]: oh sorry you're right. I'll add a comment.

File: receiver/kafkareceiver/kafka_receiver.go (hunk 4 above)
Comment (atoulme) [reply receiver/kafkareceiver/kafka_receiver.go]: yes for sure. Found out that traces were not applying auth configuration this way

File: receiver/kafkareceiver/kafka_receiver.go (hunk 5 above)
Comment (atoulme) [reply receiver/kafkareceiver/kafka_receiver.go:143]: done

File: receiver/kafkareceiver/kafka_receiver.go [hunk 6]
```diff
@@ -136,6 +139,13 @@ func (c *kafkaTracesConsumer) Start(_ context.Context, host component.Host) erro
 	if err != nil {
 		return err
 	}
+	// consumerGroup is set in tests.
```
Comment (bogdandrutu) [inline receiver/kafkareceiver/kafka_receiver.go:142]: // consumerGroup may be set in tests to inject fake implementation.

Comment (bogdandrutu) [inline receiver/kafkareceiver/kafka_receiver.go:142]: In all the cases.

File: receiver/kafkareceiver/kafka_receiver.go [hunk 7]
```diff
@@ -83,46 +86,46 @@ func newTracesReceiver(config Config, set receiver.CreateSettings, unmarshaler T
...
+	saramaConfig.Metadata.Retry.Max = config.Metadata.Retry.Max
+	saramaConfig.Metadata.Retry.Backoff = config.Metadata.Retry.Backoff
+	saramaConfig.Consumer.Offsets.AutoCommit.Enable = config.AutoCommit.Enable
+	saramaConfig.Consumer.Offsets.AutoCommit.Interval = config.AutoCommit.Interval
 	if initialOffset, err := toSaramaInitialOffset(config.InitialOffset); err == nil {
-		c.Consumer.Offsets.Initial = initialOffset
+		saramaConfig.Consumer.Offsets.Initial = initialOffset
 	} else {
 		return nil, err
 	}
```
Comment (bogdandrutu) [inline receiver/kafkareceiver/kafka_receiver.go:114]:
```golang
	var err error 
	if c.Consumer.Offsets.Initial, err = toSaramaInitialOffset(config.InitialOffset); err != nil {
		return nil, err
	}
```

Same everywhere.

File: receiver/kafkareceiver/kafka_receiver.go (hunk 6 above)
Comment (atoulme) [reply receiver/kafkareceiver/kafka_receiver.go]: done

File: receiver/kafkareceiver/kafka_receiver.go (hunk 7 above)
Comment (atoulme) [reply receiver/kafkareceiver/kafka_receiver.go]: fixed as well

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**